pkg/mark/markdown.go

```diff
@@ -90,3 +90,13 @@ func CompileMarkdown(
 
 	return string(html)
 }
+
+// dropH1Markdown will drop leading H1 headings to prevent
+// duplication of or conflict with page titles.
+func DropH1Markdown(
+	markdown []byte,
+) []byte {
+	h1 := regexp.MustCompile(`^#[^#].*\n`)
```

Comment:
The current implementation will drop all h1 headings from the entire document, is it expected? I thought you wanted to remove only the first h1 heading.

Reply:
I believe the `^` on a non `/m` regexp will only match the beginning of the document rather than the beginning of the individual line. This makes context (probably unnecessarily) important – when passing the whole markdown document from [main.go](https://github.com/kovetskiy/mark/blob/master/main.go#L135), this works as intended. If it were to be used from [meta.go](https://github.com/kovetskiy/mark/blob/master/pkg/mark/meta.go#L41) inside of the `scanner.Scan()` loop, this would have undesirable effects (like stripping comments from bash codeblocks, etc.)

`DropDocumentLeadingH1()` seems a bit verbose, but probably warranted given the potential side effects. I'll also make this clearer in comments. Let me know if you have another suggestion.